// angular-4-data-table/src/components/pagination.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/forms';
import * as i2 from '@angular/common';
import * as i3 from 'angular-4-data-table/src/utils/min';
import * as i4 from 'angular-4-data-table/src/components/pagination.component';
import * as i5 from 'angular-4-data-table/src/components/table.component';
const styles_DataTablePagination:any[] = ['.pagination-box[_ngcontent-%COMP%] {\n    position: relative;\n    margin-top: -10px;\n}\n.pagination-range[_ngcontent-%COMP%] {\n    margin-top: 7px;\n    margin-left: 3px;\n    display: inline-block;\n}\n.pagination-controllers[_ngcontent-%COMP%] {\n    float: right;\n}\n.pagination-controllers[_ngcontent-%COMP%]   input[_ngcontent-%COMP%] {\n    min-width: 60px;\n    \n    text-align: right;\n}\n\n.pagination-limit[_ngcontent-%COMP%] {\n    margin-right: 25px;\n    display: inline-table;\n    width: 150px;\n}\n.pagination-pages[_ngcontent-%COMP%] {\n    display: inline-block;\n}\n.pagination-page[_ngcontent-%COMP%] {\n    width: 110px;\n    display: inline-table;\n}\n.pagination-box[_ngcontent-%COMP%]   button[_ngcontent-%COMP%] {\n    outline: none !important;\n}\n.pagination-prevpage[_ngcontent-%COMP%], .pagination-nextpage[_ngcontent-%COMP%], .pagination-firstpage[_ngcontent-%COMP%], .pagination-lastpage[_ngcontent-%COMP%] {\n    vertical-align: top;\n}\n.pagination-reload[_ngcontent-%COMP%] {\n    color: gray;\n    font-size: 12px;\n}'];
export const RenderType_DataTablePagination:i0.RendererType2 = i0.ɵcrt({encapsulation:0,
    styles:styles_DataTablePagination,data:{}});
function View_DataTablePagination_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),9,'div',[['class',
      'pagination-range']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n        ',':\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[8,'textContent',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n        -\n        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          2,'span',([] as any[]),[[8,'textContent',0]],(null as any),(null as any),
          (null as any),(null as any))),i0.ɵpad(2),i0.ɵppd(1),(_l()(),i0.ɵted((null as any),
          ['\n        /\n        '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          0,'span',([] as any[]),[[8,'textContent',0]],(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n    ']))],
      (null as any),(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = _co.dataTable.translations.paginationRange;
        _ck(_v,1,0,currVal_0);
        const currVal_1:any = (_co.dataTable.offset + 1);
        _ck(_v,2,0,currVal_1);
        const currVal_2:any = i0.ɵunv(_v,4,0,_ck(_v,6,0,i0.ɵnov((<any>_v.parent),0),
            _ck(_v,5,0,(_co.dataTable.offset + _co.dataTable.limit),_co.dataTable.itemCount)));
        _ck(_v,4,0,currVal_2);
        const currVal_3:any = _co.dataTable.itemCount;
        _ck(_v,8,0,currVal_3);
      });
}
function View_DataTablePagination_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),15,'div',[['class',
      'pagination-limit']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),12,'div',[['class','input-group']],(null as any),
      (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
      ['\n                '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',
      [['class','input-group-addon']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['',':'])),(_l()(),i0.ɵted((null as any),
      ['\n                '])),(_l()(),i0.ɵeld(0,[['limitInput',1]],(null as any),
      6,'input',[['class','form-control'],['min','1'],['step','1'],['type','number']],
      [[2,'ng-untouched',(null as any)],[2,'ng-touched',(null as any)],[2,'ng-pristine',
          (null as any)],[2,'ng-dirty',(null as any)],[2,'ng-valid',(null as any)],
          [2,'ng-invalid',(null as any)],[2,'ng-pending',(null as any)]],[[(null as any),
          'blur'],[(null as any),'keyup.enter'],[(null as any),'keyup.esc'],[(null as any),
          'input'],[(null as any),'compositionstart'],[(null as any),'compositionend'],
          [(null as any),'change']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('input' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,8)._handleInput($event.target.value)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,8).onTouched()) !== false);
          ad = (pd_1 && ad);
        }
        if (('compositionstart' === en)) {
          const pd_2:any = ((<any>i0.ɵnov(_v,8)._compositionStart()) !== false);
          ad = (pd_2 && ad);
        }
        if (('compositionend' === en)) {
          const pd_3:any = ((<any>i0.ɵnov(_v,8)._compositionEnd($event.target.value)) !== false);
          ad = (pd_3 && ad);
        }
        if (('change' === en)) {
          const pd_4:any = ((<any>i0.ɵnov(_v,9).onChange($event.target.value)) !== false);
          ad = (pd_4 && ad);
        }
        if (('input' === en)) {
          const pd_5:any = ((<any>i0.ɵnov(_v,9).onChange($event.target.value)) !== false);
          ad = (pd_5 && ad);
        }
        if (('blur' === en)) {
          const pd_6:any = ((<any>i0.ɵnov(_v,9).onTouched()) !== false);
          ad = (pd_6 && ad);
        }
        if (('blur' === en)) {
          const pd_7:any = ((<any>(_co.limit = i0.ɵnov(_v,7).value)) !== false);
          ad = (pd_7 && ad);
        }
        if (('keyup.enter' === en)) {
          const pd_8:any = ((<any>(_co.limit = i0.ɵnov(_v,7).value)) !== false);
          ad = (pd_8 && ad);
        }
        if (('keyup.esc' === en)) {
          const pd_9:any = ((<any>(i0.ɵnov(_v,7).value = _co.limit)) !== false);
          ad = (pd_9 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i1.DefaultValueAccessor,
      [i0.Renderer2,i0.ElementRef,[2,i1.COMPOSITION_BUFFER_MODE]],(null as any),(null as any)),
      i0.ɵdid(16384,(null as any),0,i1.ɵbc,[i0.Renderer2,i0.ElementRef],(null as any),
          (null as any)),i0.ɵprd(1024,(null as any),i1.NG_VALUE_ACCESSOR,(p0_0:any,
          p1_0:any) => {
        return [p0_0,p1_0];
      },[i1.DefaultValueAccessor,i1.ɵbc]),i0.ɵdid(671744,(null as any),0,i1.NgModel,
          [[8,(null as any)],[8,(null as any)],[8,(null as any)],[2,i1.NG_VALUE_ACCESSOR]],
          {model:[0,'model']},(null as any)),i0.ɵprd(2048,(null as any),i1.NgControl,
          (null as any),[i1.NgModel]),i0.ɵdid(16384,(null as any),0,i1.NgControlStatus,
          [i1.NgControl],(null as any),(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵted((null as any),['\n        ']))],(_ck,
      _v) => {
    var _co:any = _v.component;
    const currVal_8:any = _co.limit;
    _ck(_v,11,0,currVal_8);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.dataTable.translations.paginationLimit;
    _ck(_v,5,0,currVal_0);
    const currVal_1:any = i0.ɵnov(_v,13).ngClassUntouched;
    const currVal_2:any = i0.ɵnov(_v,13).ngClassTouched;
    const currVal_3:any = i0.ɵnov(_v,13).ngClassPristine;
    const currVal_4:any = i0.ɵnov(_v,13).ngClassDirty;
    const currVal_5:any = i0.ɵnov(_v,13).ngClassValid;
    const currVal_6:any = i0.ɵnov(_v,13).ngClassInvalid;
    const currVal_7:any = i0.ɵnov(_v,13).ngClassPending;
    _ck(_v,7,0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
  });
}
function View_DataTablePagination_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),20,'div',[['class',
      'pagination-page']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵeld(0,(null as any),(null as any),17,'div',[['class','input-group']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
      ['\n                    '])),(_l()(),i0.ɵeld(0,[['pageInput',1]],(null as any),
      6,'input',[['class','form-control'],['min','1'],['step','1'],['type','number']],
      [[8,'max',0],[2,'ng-untouched',(null as any)],[2,'ng-touched',(null as any)],
          [2,'ng-pristine',(null as any)],[2,'ng-dirty',(null as any)],[2,'ng-valid',
              (null as any)],[2,'ng-invalid',(null as any)],[2,'ng-pending',(null as any)]],
      [[(null as any),'blur'],[(null as any),'keyup.enter'],[(null as any),'keyup.esc'],
          [(null as any),'input'],[(null as any),'compositionstart'],[(null as any),
              'compositionend'],[(null as any),'change']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('input' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,5)._handleInput($event.target.value)) !== false);
          ad = (pd_0 && ad);
        }
        if (('blur' === en)) {
          const pd_1:any = ((<any>i0.ɵnov(_v,5).onTouched()) !== false);
          ad = (pd_1 && ad);
        }
        if (('compositionstart' === en)) {
          const pd_2:any = ((<any>i0.ɵnov(_v,5)._compositionStart()) !== false);
          ad = (pd_2 && ad);
        }
        if (('compositionend' === en)) {
          const pd_3:any = ((<any>i0.ɵnov(_v,5)._compositionEnd($event.target.value)) !== false);
          ad = (pd_3 && ad);
        }
        if (('change' === en)) {
          const pd_4:any = ((<any>i0.ɵnov(_v,6).onChange($event.target.value)) !== false);
          ad = (pd_4 && ad);
        }
        if (('input' === en)) {
          const pd_5:any = ((<any>i0.ɵnov(_v,6).onChange($event.target.value)) !== false);
          ad = (pd_5 && ad);
        }
        if (('blur' === en)) {
          const pd_6:any = ((<any>i0.ɵnov(_v,6).onTouched()) !== false);
          ad = (pd_6 && ad);
        }
        if (('blur' === en)) {
          const pd_7:any = ((<any>(_co.page = i0.ɵnov(_v,4).value)) !== false);
          ad = (pd_7 && ad);
        }
        if (('keyup.enter' === en)) {
          const pd_8:any = ((<any>(_co.page = i0.ɵnov(_v,4).value)) !== false);
          ad = (pd_8 && ad);
        }
        if (('keyup.esc' === en)) {
          const pd_9:any = ((<any>(i0.ɵnov(_v,4).value = _co.page)) !== false);
          ad = (pd_9 && ad);
        }
        return ad;
      },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i1.DefaultValueAccessor,
      [i0.Renderer2,i0.ElementRef,[2,i1.COMPOSITION_BUFFER_MODE]],(null as any),(null as any)),
      i0.ɵdid(16384,(null as any),0,i1.ɵbc,[i0.Renderer2,i0.ElementRef],(null as any),
          (null as any)),i0.ɵprd(1024,(null as any),i1.NG_VALUE_ACCESSOR,(p0_0:any,
          p1_0:any) => {
        return [p0_0,p1_0];
      },[i1.DefaultValueAccessor,i1.ɵbc]),i0.ɵdid(671744,(null as any),0,i1.NgModel,
          [[8,(null as any)],[8,(null as any)],[8,(null as any)],[2,i1.NG_VALUE_ACCESSOR]],
          {model:[0,'model']},(null as any)),i0.ɵprd(2048,(null as any),i1.NgControl,
          (null as any),[i1.NgModel]),i0.ɵdid(16384,(null as any),0,i1.NgControlStatus,
          [i1.NgControl],(null as any),(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          6,'div',[['class','input-group-addon']],(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['/'])),(_l()(),i0.ɵted((null as any),['\n                        '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[8,'textContent',
              0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),
          i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵted((null as any),
          ['\n                '])),(_l()(),i0.ɵted((null as any),['\n            ']))],
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_8:any = _co.page;
        _ck(_v,8,0,currVal_8);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = i0.ɵinlineInterpolate(1,'',_co.maxPage,'');
        const currVal_1:any = i0.ɵnov(_v,10).ngClassUntouched;
        const currVal_2:any = i0.ɵnov(_v,10).ngClassTouched;
        const currVal_3:any = i0.ɵnov(_v,10).ngClassPristine;
        const currVal_4:any = i0.ɵnov(_v,10).ngClassDirty;
        const currVal_5:any = i0.ɵnov(_v,10).ngClassValid;
        const currVal_6:any = i0.ɵnov(_v,10).ngClassInvalid;
        const currVal_7:any = i0.ɵnov(_v,10).ngClassPending;
        _ck(_v,4,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4,currVal_5,currVal_6,
            currVal_7);
        const currVal_9:any = _co.dataTable.lastPage;
        _ck(_v,17,0,currVal_9);
      });
}
function View_DataTablePagination_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default hasPrevious']],[[8,'disabled',0]],[[(null as any),'click']],
      (_v,en,$event) => {
        var ad:boolean = true;
        if (('click' === en)) {
          const pd_0:any = ((<any>false) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['...']))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = true;
        _ck(_v,0,0,currVal_0);
      });
}
function View_DataTablePagination_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default']],[[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>(_co.page = _v.context.$implicit)) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['','']))],(null as any),
      (_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = (_v.context.$implicit == _co.page);
        _ck(_v,0,0,currVal_0);
        const currVal_1:any = _v.context.$implicit;
        _ck(_v,1,0,currVal_1);
      });
}
function View_DataTablePagination_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',[['class',
      'pagination-page']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTablePagination_6)),
      i0.ɵdid(802816,(null as any),0,i2.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.createPageRange(_co.maxPage,_co.page);
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_DataTablePagination_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class',
      'btn btn-default hasNext']],[[8,'disabled',0]],[[(null as any),'click']],(_v,
      en,$event) => {
    var ad:boolean = true;
    if (('click' === en)) {
      const pd_0:any = ((<any>false) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['...']))],(null as any),
      (_ck,_v) => {
        const currVal_0:any = true;
        _ck(_v,0,0,currVal_0);
      });
}
export function View_DataTablePagination_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵpid(0,i3.MinPipe,([] as any[])),(_l()(),i0.ɵted((null as any),
      ['\n'])),(_l()(),i0.ɵeld(0,(null as any),(null as any),37,'div',[['class','pagination-box']],
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_DataTablePagination_1)),i0.ɵdid(16384,(null as any),0,i2.NgIf,
      [i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),
      i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
      31,'div',[['class','pagination-controllers']],(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTablePagination_2)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),25,'div',[['class',' pagination-pages']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'button',[['class','btn btn-default pagination-firstpage']],
          [[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i4.DataTablePagination = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.pageFirst()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['«'])),(_l()(),
          i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'button',[['class','btn btn-default pagination-prevpage']],
          [[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i4.DataTablePagination = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.pageBack()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['‹'])),(_l()(),
          i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_DataTablePagination_3)),i0.ɵdid(16384,
          (null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTablePagination_4)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTablePagination_5)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTablePagination_7)),
      i0.ɵdid(16384,(null as any),0,i2.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'button',[['class','btn btn-default pagination-nextpage']],
          [[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i4.DataTablePagination = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.pageForward()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['›'])),(_l()(),
          i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),1,'button',[['class','btn btn-default pagination-lastpage']],
          [[8,'disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i4.DataTablePagination = _v.component;
            if (('click' === en)) {
              const pd_0:any = ((<any>_co.pageLast()) !== false);
              ad = (pd_0 && ad);
            }
            return ad;
          },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['»'])),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵted((null as any),['\n']))],
      (_ck,_v) => {
        var _co:i4.DataTablePagination = _v.component;
        const currVal_0:any = _co.show_range;
        _ck(_v,5,0,currVal_0);
        const currVal_1:any = _co.show_limit;
        _ck(_v,10,0,currVal_1);
        const currVal_4:any = _co.show_input;
        _ck(_v,21,0,currVal_4);
        const currVal_5:any = _co.hasPrevious(_co.maxPage,_co.page);
        _ck(_v,24,0,currVal_5);
        const currVal_6:any = _co.show_numbers;
        _ck(_v,27,0,currVal_6);
        const currVal_7:any = _co.hasNext(_co.maxPage,_co.page);
        _ck(_v,30,0,currVal_7);
      },(_ck,_v) => {
        var _co:i4.DataTablePagination = _v.component;
        const currVal_2:any = (_co.dataTable.offset <= 0);
        _ck(_v,14,0,currVal_2);
        const currVal_3:any = (_co.dataTable.offset <= 0);
        _ck(_v,17,0,currVal_3);
        const currVal_8:any = ((_co.dataTable.offset + _co.dataTable.limit) >= _co.dataTable.itemCount);
        _ck(_v,32,0,currVal_8);
        const currVal_9:any = ((_co.dataTable.offset + _co.dataTable.limit) >= _co.dataTable.itemCount);
        _ck(_v,35,0,currVal_9);
      });
}
export function View_DataTablePagination_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'data-table-pagination',
      ([] as any[]),(null as any),(null as any),(null as any),View_DataTablePagination_0,
      RenderType_DataTablePagination)),i0.ɵdid(49152,(null as any),0,i4.DataTablePagination,
      [i5.DataTable],(null as any),(null as any))],(null as any),(null as any));
}
export const DataTablePaginationNgFactory:i0.ComponentFactory<i4.DataTablePagination> = i0.ɵccf('data-table-pagination',
    i4.DataTablePagination,View_DataTablePagination_Host_0,{show_range:'show_range',
        show_limit:'show_limit',show_input:'show_input',show_numbers:'show_numbers'},
    {},([] as any[]));
