// angular-4-data-table/src/components/table.component.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from './header.component.ngfactory';
import * as i2 from 'angular-4-data-table/src/components/header.component';
import * as i3 from 'angular-4-data-table/src/components/table.component';
import * as i4 from '@angular/common';
import * as i5 from 'angular-4-data-table/src/utils/hide';
import * as i6 from './row.component.ngfactory';
import * as i7 from 'angular-4-data-table/src/components/row.component';
import * as i8 from './pagination.component.ngfactory';
import * as i9 from 'angular-4-data-table/src/components/pagination.component';
import * as i10 from 'angular-4-data-table/src/utils/px';
import * as i11 from '@angular/forms';
const styles_DataTable:any[] = ['[_nghost-%COMP%]     .data-table.table > tbody+tbody {\n    border-top: none;\n}\n[_nghost-%COMP%]     .data-table.table td {\n    vertical-align: middle;\n}\n\n[_nghost-%COMP%]     .data-table > thead > tr > th, [_nghost-%COMP%]     .data-table > tbody > tr > td {\n	overflow: hidden;\n}\n\n\n[_nghost-%COMP%]     .row-odd {\n    background-color: #F6F6F6;\n}\n[_nghost-%COMP%]     .row-even {\n}\n\n.data-table[_ngcontent-%COMP%]   .substitute-rows[_ngcontent-%COMP%]    > tr[_ngcontent-%COMP%]:hover, [_nghost-%COMP%]     .data-table .data-table-row:hover {\n    background-color: #ECECEC;\n}\n\n\n.data-table[_ngcontent-%COMP%] {\n    box-shadow: 0 0 15px rgb(236, 236, 236);\n    table-layout: fixed;\n}\n\n\n\n.column-header[_ngcontent-%COMP%] {\n    position: relative;\n}\n.expand-column-header[_ngcontent-%COMP%] {\n	width: 50px;\n}\n.select-column-header[_ngcontent-%COMP%] {\n	width: 50px;\n	text-align: center;\n}\n.index-column-header[_ngcontent-%COMP%] {\n	width: 40px;\n}\n.column-header.sortable[_ngcontent-%COMP%] {\n    cursor: pointer;\n}\n.column-header[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%] {\n	float: right;\n}\n.column-header.resizable[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%] {\n    margin-right: 8px;\n}\n.column-header[_ngcontent-%COMP%]   .column-sort-icon[_ngcontent-%COMP%]   .column-sortable-icon[_ngcontent-%COMP%] {\n    color: lightgray;\n}\n.column-header[_ngcontent-%COMP%]   .column-resize-handle[_ngcontent-%COMP%] {\n    position: absolute;\n    top: 0;\n    right: 0;\n    margin: 0;\n    padding: 0;\n    width: 8px;\n    height: 100%;\n    cursor: col-resize;\n}\n\n\n\n.data-table-box[_ngcontent-%COMP%] {\n    position: relative;\n}\n\n.loading-cover[_ngcontent-%COMP%] {\n   position: absolute;\n   width: 100%;\n   height: 100%;\n   background-color: rgba(255, 255, 255, 0.3);\n   top: 0;\n}'];
export const RenderType_DataTable:i0.RendererType2 = i0.ɵcrt({encapsulation:0,styles:styles_DataTable,
    data:{}});
function View_DataTable_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'data-table-header',
      ([] as any[]),(null as any),[['document','click']],(_v,en,$event) => {
        var ad:boolean = true;
        if (('document:click' === en)) {
          const pd_0:any = ((<any>i0.ɵnov(_v,1)._closeSelector()) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },i1.View_DataTableHeader_0,i1.RenderType_DataTableHeader)),i0.ɵdid(49152,(null as any),
      0,i2.DataTableHeader,[i3.DataTable],(null as any),(null as any))],(null as any),
      (null as any));
}
function View_DataTable_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),
      [[8,'textContent',0]],(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(_ck,_v) => {
        const currVal_0:any = (<any>_v.parent).context.$implicit.header;
        _ck(_v,0,0,currVal_0);
      });
}
function View_DataTable_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(16777216,(null as any),(null as any),2,'span',
      ([] as any[]),(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(540672,(null as any),0,i4.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutlet:[0,
          'ngTemplateOutlet'],ngOutletContext:[1,'ngOutletContext']},(null as any)),
      i0.ɵpod({column:0}),(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(_ck,
      _v) => {
    const currVal_0:any = (<any>_v.parent).context.$implicit.headerTemplate;
    const currVal_1:any = _ck(_v,2,0,(<any>_v.parent).context.$implicit);
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_DataTable_5(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),12,'span',[['class',
      'column-sort-icon']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'span',[['class','glyphicon glyphicon-sort column-sortable-icon']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,(null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                            '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),6,'span',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
          (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                                 '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),2,'span',[['class','glyphicon']],(null as any),
              (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(278528,
          (null as any),0,i4.NgClass,[i0.IterableDiffers,i0.KeyValueDiffers,i0.ElementRef,
              i0.Renderer],{klass:[0,'klass'],ngClass:[1,'ngClass']},(null as any)),
      i0.ɵpod({'glyphicon-triangle-top':0,'glyphicon-triangle-bottom':1}),(_l()(),
          i0.ɵted((null as any),['\n                            '])),(_l()(),i0.ɵted((null as any),
          ['\n                        ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = ((<any>_v.parent).context.$implicit.property === _co.sortBy);
    _ck(_v,3,0,currVal_0);
    const currVal_1:any = ((<any>_v.parent).context.$implicit.property !== _co.sortBy);
    _ck(_v,6,0,currVal_1);
    const currVal_2:any = 'glyphicon';
    const currVal_3:any = _ck(_v,10,0,!_co.sortAsc,_co.sortAsc);
    _ck(_v,9,0,currVal_2,currVal_3);
  },(null as any));
}
function View_DataTable_6(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',[['class',
      'column-resize-handle']],(null as any),[[(null as any),'mousedown']],(_v,en,
      $event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('mousedown' === en)) {
      const pd_0:any = ((<any>_co.resizeColumnStart($event,(<any>_v.parent).context.$implicit,
          i0.ɵnov((<any>_v.parent),0))) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },(null as any),(null as any)))],(null as any),(null as any));
}
function View_DataTable_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,[['th',1]],(null as any),16,'th',[['class','column-header'],
      ['scope','col']],[[2,'sortable',(null as any)],[2,'resizable',(null as any)],
      [4,'width',(null as any)],[1,'aria-sort',0],[1,'tabindex',0]],[[(null as any),
      'click'],[(null as any),'keydown.enter'],[(null as any),'keydown.space']],(_v,
      en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('click' === en)) {
      const pd_0:any = ((<any>_co.headerClicked(_v.context.$implicit,$event)) !== false);
      ad = (pd_0 && ad);
    }
    if (('keydown.enter' === en)) {
      const pd_1:any = ((<any>_co.headerClicked(_v.context.$implicit,$event)) !== false);
      ad = (pd_1 && ad);
    }
    if (('keydown.space' === en)) {
      const pd_2:any = ((<any>_co.headerClicked(_v.context.$implicit,$event)) !== false);
      ad = (pd_2 && ad);
    }
    return ad;
  },(null as any),(null as any))),i0.ɵdid(278528,(null as any),0,i4.NgClass,[i0.IterableDiffers,
      i0.KeyValueDiffers,i0.ElementRef,i0.Renderer],{klass:[0,'klass'],ngClass:[1,
      'ngClass']},(null as any)),i0.ɵdid(16384,(null as any),0,i5.Hide,[i0.ElementRef,
      i0.Renderer],{hide:[0,'hide']},(null as any)),i0.ɵppd(1),(_l()(),i0.ɵted((null as any),
      ['\n                        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_DataTable_3)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n                        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_DataTable_4)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n                        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_DataTable_5)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n                        '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
      1,(null as any),View_DataTable_6)),i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,
      i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),
      ['\n                    ']))],(_ck,_v) => {
    const currVal_5:any = 'column-header';
    const currVal_6:any = _v.context.$implicit.styleClassObject;
    _ck(_v,1,0,currVal_5,currVal_6);
    const currVal_7:boolean = !_v.context.$implicit.visible;
    _ck(_v,2,0,currVal_7);
    const currVal_8:boolean = !_v.context.$implicit.headerTemplate;
    _ck(_v,6,0,currVal_8);
    const currVal_9:any = _v.context.$implicit.headerTemplate;
    _ck(_v,9,0,currVal_9);
    const currVal_10:any = _v.context.$implicit.sortable;
    _ck(_v,12,0,currVal_10);
    const currVal_11:any = _v.context.$implicit.resizable;
    _ck(_v,15,0,currVal_11);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _v.context.$implicit.sortable;
    const currVal_1:any = _v.context.$implicit.resizable;
    const currVal_2:any = i0.ɵunv(_v,0,2,_ck(_v,3,0,i0.ɵnov((<any>_v.parent),0),_v.context.$implicit.width));
    const currVal_3:any = (_v.context.$implicit.sortable? ((_v.context.$implicit.property === _co.sortBy)? (_co.sortAsc? 'ascending': 'descending'): 'none'): (null as any));
    const currVal_4:any = (_v.context.$implicit.sortable? '0': (null as any));
    _ck(_v,0,0,currVal_0,currVal_1,currVal_2,currVal_3,currVal_4);
  });
}
function View_DataTable_7(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'tbody',[['class',
      'data-table-row-wrapper'],['dataTableRow','']],(null as any),[[(null as any),
      'selectedChange']],(_v,en,$event) => {
    var ad:boolean = true;
    var _co:any = _v.component;
    if (('selectedChange' === en)) {
      const pd_0:any = ((<any>_co.onRowSelectChanged(i0.ɵnov(_v,1))) !== false);
      ad = (pd_0 && ad);
    }
    return ad;
  },i6.View_DataTableRow_0,i6.RenderType_DataTableRow)),i0.ɵdid(180224,[[1,4],['row',
      4]],0,i7.DataTableRow,[i3.DataTable],{item:[0,'item'],index:[1,'index']},{selectedChange:'selectedChange'}),
      (_l()(),i0.ɵted((null as any),['\n            ']))],(_ck,_v) => {
    const currVal_0:any = _v.context.$implicit;
    const currVal_1:any = _v.context.index;
    _ck(_v,1,0,currVal_0,currVal_1);
  },(null as any));
}
function View_DataTable_8(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),7,'tbody',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),(_l()(),
      i0.ɵted((null as any),['\n                '])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),4,'tr',([] as any[]),(null as any),(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),[[1,'colspan',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['',''])),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵted((null as any),['\n            ']))],(null as any),(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.columnCount;
    _ck(_v,4,0,currVal_0);
    const currVal_1:any = _co.noDataMessage;
    _ck(_v,5,0,currVal_1);
  });
}
function View_DataTable_11(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'td',([] as any[]),
      (null as any),(null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
      (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                ']))],(_ck,_v) => {
    const currVal_0:boolean = !_v.context.$implicit.visible;
    _ck(_v,1,0,currVal_0);
  },(null as any));
}
function View_DataTable_10(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),13,'tr',([] as any[]),
      [[2,'row-odd',(null as any)],[2,'row-even',(null as any)]],(null as any),(null as any),
      (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
          (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),2,'td',([] as any[]),(null as any),(null as any),
          (null as any),(null as any),(null as any))),i0.ɵdid(16384,(null as any),
          0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),(_l()(),
          i0.ɵted((null as any),[' '])),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),1,'td',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
          (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵand(16777216,
          (null as any),(null as any),1,(null as any),View_DataTable_11)),i0.ɵdid(802816,
          (null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
          {ngForOf:[0,'ngForOf']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_2:boolean = !_co.expandColumnVisible;
    _ck(_v,3,0,currVal_2);
    const currVal_3:boolean = !_co.indexColumnVisible;
    _ck(_v,6,0,currVal_3);
    const currVal_4:boolean = !_co.selectColumnVisible;
    _ck(_v,10,0,currVal_4);
    const currVal_5:any = _co.columns;
    _ck(_v,13,0,currVal_5);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = (((_v.context.index + _co.items.length) % 2) === 0);
    const currVal_1:any = (((_v.context.index + _co.items.length) % 2) === 1);
    _ck(_v,0,0,currVal_0,currVal_1);
  });
}
function View_DataTable_9(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'tbody',[['class',
      'substitute-rows']],(null as any),(null as any),(null as any),(null as any),
      (null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
      i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTable_10)),
      i0.ɵdid(802816,(null as any),0,i4.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,
          i0.IterableDiffers],{ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),
          ['\n            ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.substituteItems;
    _ck(_v,3,0,currVal_0);
  },(null as any));
}
function View_DataTable_12(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),0,'div',[['class',
      'loading-cover']],(null as any),(null as any),(null as any),(null as any),(null as any)))],
      (null as any),(null as any));
}
function View_DataTable_13(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),1,'data-table-pagination',
      ([] as any[]),(null as any),(null as any),(null as any),i8.View_DataTablePagination_0,
      i8.RenderType_DataTablePagination)),i0.ɵdid(49152,(null as any),0,i9.DataTablePagination,
      [i3.DataTable],{show_range:[0,'show_range'],show_limit:[1,'show_limit'],show_input:[2,
          'show_input'],show_numbers:[3,'show_numbers']},(null as any))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = _co.pagination_range;
    const currVal_1:any = _co.pagination_limit;
    const currVal_2:any = _co.pagination_input;
    const currVal_3:any = _co.pagination_numbers;
    _ck(_v,1,0,currVal_0,currVal_1,currVal_2,currVal_3);
  },(null as any));
}
export function View_DataTable_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[i0.ɵpid(0,i10.PixelConverter,([] as any[])),i0.ɵqud(671088640,
      1,{rows:1}),(_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵeld(0,(null as any),
      (null as any),55,'div',[['class','data-table-wrapper']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTable_1)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n\n    '])),(_l()(),
          i0.ɵeld(0,(null as any),(null as any),46,'div',[['class','data-table-box']],
              (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),40,'table',[['class','table table-condensed table-bordered data-table']],
          (null as any),(null as any),(null as any),(null as any),(null as any))),
      (_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵeld(0,(null as any),
          (null as any),28,'thead',([] as any[]),(null as any),(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n                '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),25,'tr',([] as any[]),(null as any),
          (null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵeld(0,(null as any),(null as any),
          2,'th',[['class','expand-column-header'],['scope','col']],(null as any),
          (null as any),(null as any),(null as any),(null as any))),i0.ɵdid(16384,
          (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵeld(0,
          (null as any),(null as any),4,'th',[['class','index-column-header'],['scope',
              'col']],(null as any),(null as any),(null as any),(null as any),(null as any))),
      i0.ɵdid(16384,(null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),0,'span',([] as any[]),[[8,'textContent',
          0]],(null as any),(null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵted((null as any),['\n                    '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),10,'th',[['class','select-column-header'],
          ['scope','col']],(null as any),(null as any),(null as any),(null as any),
          (null as any))),i0.ɵdid(16384,(null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],
          {hide:[0,'hide']},(null as any)),(_l()(),i0.ɵted((null as any),['\n                        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),6,'input',[['type','checkbox']],
          [[1,'aria-label',0],[2,'ng-untouched',(null as any)],[2,'ng-touched',(null as any)],
              [2,'ng-pristine',(null as any)],[2,'ng-dirty',(null as any)],[2,'ng-valid',
                  (null as any)],[2,'ng-invalid',(null as any)],[2,'ng-pending',(null as any)]],
          [[(null as any),'ngModelChange'],[(null as any),'change'],[(null as any),
              'blur']],(_v,en,$event) => {
            var ad:boolean = true;
            var _co:i3.DataTable = _v.component;
            if (('change' === en)) {
              const pd_0:any = ((<any>i0.ɵnov(_v,29).onChange($event.target.checked)) !== false);
              ad = (pd_0 && ad);
            }
            if (('blur' === en)) {
              const pd_1:any = ((<any>i0.ɵnov(_v,29).onTouched()) !== false);
              ad = (pd_1 && ad);
            }
            if (('ngModelChange' === en)) {
              const pd_2:any = ((<any>(_co.selectAllCheckbox = $event)) !== false);
              ad = (pd_2 && ad);
            }
            return ad;
          },(null as any),(null as any))),i0.ɵdid(16384,(null as any),0,i11.CheckboxControlValueAccessor,
          [i0.Renderer2,i0.ElementRef],(null as any),(null as any)),i0.ɵprd(1024,(null as any),
          i11.NG_VALUE_ACCESSOR,(p0_0:any) => {
            return [p0_0];
          },[i11.CheckboxControlValueAccessor]),i0.ɵdid(671744,(null as any),0,i11.NgModel,
          [[8,(null as any)],[8,(null as any)],[8,(null as any)],[2,i11.NG_VALUE_ACCESSOR]],
          {model:[0,'model']},{update:'ngModelChange'}),i0.ɵprd(2048,(null as any),
          i11.NgControl,(null as any),[i11.NgModel]),i0.ɵdid(16384,(null as any),0,
          i11.NgControlStatus,[i11.NgControl],(null as any),(null as any)),i0.ɵdid(16384,
          (null as any),0,i5.Hide,[i0.ElementRef,i0.Renderer],{hide:[0,'hide']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n                    '])),(_l()(),i0.ɵted((null as any),
          ['\n                    '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_DataTable_2)),i0.ɵdid(802816,(null as any),0,i4.NgForOf,
          [i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n                '])),(_l()(),
          i0.ɵted((null as any),['\n            '])),(_l()(),i0.ɵted((null as any),
          ['\n            '])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),
          1,(null as any),View_DataTable_7)),i0.ɵdid(802816,(null as any),0,i4.NgForOf,
          [i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],{ngForOf:[0,'ngForOf']},
          (null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTable_8)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n            '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTable_9)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵted((null as any),['\n        '])),(_l()(),i0.ɵand(16777216,(null as any),
          (null as any),1,(null as any),View_DataTable_12)),i0.ɵdid(16384,(null as any),
          0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,'ngIf']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵted((null as any),['\n\n    '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_DataTable_13)),
      i0.ɵdid(16384,(null as any),0,i4.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n'])),(_l()(),i0.ɵted((null as any),
          ['\n']))],(_ck,_v) => {
    var _co:i3.DataTable = _v.component;
    const currVal_0:any = _co.header;
    _ck(_v,6,0,currVal_0);
    const currVal_1:boolean = !_co.expandColumnVisible;
    _ck(_v,17,0,currVal_1);
    const currVal_2:boolean = !_co.indexColumnVisible;
    _ck(_v,20,0,currVal_2);
    const currVal_4:boolean = !_co.selectColumnVisible;
    _ck(_v,26,0,currVal_4);
    const currVal_13:any = _co.selectAllCheckbox;
    _ck(_v,31,0,currVal_13);
    const currVal_14:boolean = !_co.multiSelect;
    _ck(_v,34,0,currVal_14);
    const currVal_15:any = _co.columns;
    _ck(_v,38,0,currVal_15);
    const currVal_16:any = _co.items;
    _ck(_v,43,0,currVal_16);
    const currVal_17:any = ((_co.itemCount === 0) && _co.noDataMessage);
    _ck(_v,46,0,currVal_17);
    const currVal_18:any = (_co.pagination && _co.substituteRows);
    _ck(_v,49,0,currVal_18);
    const currVal_19:any = (_co.showReloading && _co.reloading);
    _ck(_v,53,0,currVal_19);
    const currVal_20:any = _co.pagination;
    _ck(_v,57,0,currVal_20);
  },(_ck,_v) => {
    var _co:i3.DataTable = _v.component;
    const currVal_3:any = _co.indexColumnHeader;
    _ck(_v,22,0,currVal_3);
    const currVal_5:any = _co.translations.selectAllRows;
    const currVal_6:any = i0.ɵnov(_v,33).ngClassUntouched;
    const currVal_7:any = i0.ɵnov(_v,33).ngClassTouched;
    const currVal_8:any = i0.ɵnov(_v,33).ngClassPristine;
    const currVal_9:any = i0.ɵnov(_v,33).ngClassDirty;
    const currVal_10:any = i0.ɵnov(_v,33).ngClassValid;
    const currVal_11:any = i0.ɵnov(_v,33).ngClassInvalid;
    const currVal_12:any = i0.ɵnov(_v,33).ngClassPending;
    _ck(_v,28,0,currVal_5,currVal_6,currVal_7,currVal_8,currVal_9,currVal_10,currVal_11,
        currVal_12);
  });
}
export function View_DataTable_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),3,'data-table',([] as any[]),
      (null as any),(null as any),(null as any),View_DataTable_0,RenderType_DataTable)),
      i0.ɵdid(114688,(null as any),2,i3.DataTable,([] as any[]),(null as any),(null as any)),
      i0.ɵqud(603979776,1,{columns:1}),i0.ɵqud(335544320,2,{expandTemplate:0})],(_ck,
      _v) => {
    _ck(_v,1,0);
  },(null as any));
}
export const DataTableNgFactory:i0.ComponentFactory<i3.DataTable> = i0.ɵccf('data-table',
    i3.DataTable,View_DataTable_Host_0,{items:'items',itemCount:'itemCount',headerTitle:'headerTitle',
        header:'header',pagination:'pagination',pagination_range:'pagination_range',
        pagination_limit:'pagination_limit',pagination_input:'pagination_input',pagination_numbers:'pagination_numbers',
        indexColumn:'indexColumn',indexColumnHeader:'indexColumnHeader',rowColors:'rowColors',
        rowTooltip:'rowTooltip',selectColumn:'selectColumn',multiSelect:'multiSelect',
        substituteRows:'substituteRows',expandableRows:'expandableRows',translations:'translations',
        selectOnRowClick:'selectOnRowClick',autoReload:'autoReload',showReloading:'showReloading',
        noDataMessage:'noDataMessage',sortBy:'sortBy',sortAsc:'sortAsc',offset:'offset',
        limit:'limit',page:'page'},{reload:'reload',rowClick:'rowClick',rowDoubleClick:'rowDoubleClick',
        headerClick:'headerClick',cellClick:'cellClick'},([] as any[]));
