// @ng-bootstrap/ng-bootstrap/accordion/accordion.ngfactory.ts
/**
 * @fileoverview This file is generated by the Angular template compiler.
 * Do not edit.
 * @suppress {suspiciousCode,uselessCode,missingProperties,missingOverride}
 */
 /* tslint:disable */


import * as i0 from '@angular/core';
import * as i1 from '@angular/common';
import * as i2 from '@ng-bootstrap/ng-bootstrap/accordion/accordion';
import * as i3 from '@ng-bootstrap/ng-bootstrap/accordion/accordion-config';
const styles_NgbAccordion:any[] = ([] as any[]);
export const RenderType_NgbAccordion:i0.RendererType2 = i0.ɵcrt({encapsulation:2,styles:styles_NgbAccordion,
    data:{}});
function View_NgbAccordion_2(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbAccordion_4(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵand(0,(null as any),(null as any),0))],(null as any),
      (null as any));
}
function View_NgbAccordion_3(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),4,'div',[['role',
      'tabpanel']],[[8,'id',0],[1,'aria-labelledby',0],[8,'className',0]],(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n             '])),
      (_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbAccordion_4)),
      i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,[i0.ViewContainerRef],{ngTemplateOutlet:[0,
          'ngTemplateOutlet']},(null as any)),(_l()(),i0.ɵted((null as any),['\n        ']))],
      (_ck,_v) => {
        const currVal_3:any = (<any>_v.parent).context.$implicit.contentTpl.templateRef;
        _ck(_v,3,0,currVal_3);
      },(_ck,_v) => {
        var _co:any = _v.component;
        const currVal_0:any = i0.ɵinlineInterpolate(1,'',(<any>_v.parent).context.$implicit.id,
            '');
        const currVal_1:any = ((<any>_v.parent).context.$implicit.id + '-header');
        const currVal_2:any = i0.ɵinlineInterpolate(1,'card-body ',(_co.isOpen((<any>_v.parent).context.$implicit.id)? 'show': (null as any)),
            '');
        _ck(_v,0,0,currVal_0,currVal_1,currVal_2);
      });
}
function View_NgbAccordion_1(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),i0.ɵeld(0,
      (null as any),(null as any),13,'div',[['class','card']],(null as any),(null as any),
      (null as any),(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n        '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),7,'div',[['role','tab']],[[8,'id',
          0],[8,'className',0],[2,'active',(null as any)]],(null as any),(null as any),
          (null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n          '])),
      (_l()(),i0.ɵeld(0,(null as any),(null as any),4,'a',[['href','']],[[2,'text-muted',
          (null as any)],[1,'tabindex',0],[1,'aria-expanded',0],[1,'aria-controls',
          0],[1,'aria-disabled',0]],[[(null as any),'click']],(_v,en,$event) => {
        var ad:boolean = true;
        var _co:any = _v.component;
        if (('click' === en)) {
          const pd_0:any = ((<any>!!_co.toggle(_v.context.$implicit.id)) !== false);
          ad = (pd_0 && ad);
        }
        return ad;
      },(null as any),(null as any))),(_l()(),i0.ɵted((null as any),['\n            ',
          ''])),(_l()(),i0.ɵand(16777216,(null as any),(null as any),1,(null as any),
          View_NgbAccordion_2)),i0.ɵdid(540672,(null as any),0,i1.NgTemplateOutlet,
          [i0.ViewContainerRef],{ngTemplateOutlet:[0,'ngTemplateOutlet']},(null as any)),
      (_l()(),i0.ɵted((null as any),['\n          '])),(_l()(),i0.ɵted((null as any),
          ['\n        '])),(_l()(),i0.ɵted((null as any),['\n        '])),(_l()(),
          i0.ɵand(16777216,(null as any),(null as any),1,(null as any),View_NgbAccordion_3)),
      i0.ɵdid(16384,(null as any),0,i1.NgIf,[i0.ViewContainerRef,i0.TemplateRef],{ngIf:[0,
          'ngIf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n      '])),(_l()(),
          i0.ɵted((null as any),['\n    ']))],(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_9:any = ((_v.context.$implicit.titleTpl == null)? (null as any): _v.context.$implicit.titleTpl.templateRef);
    _ck(_v,8,0,currVal_9);
    const currVal_10:boolean = (!_co.destroyOnHide || _co.isOpen(_v.context.$implicit.id));
    _ck(_v,13,0,currVal_10);
  },(_ck,_v) => {
    var _co:any = _v.component;
    const currVal_0:any = i0.ɵinlineInterpolate(1,'',_v.context.$implicit.id,'-header');
    const currVal_1:any = ('card-header ' + (_v.context.$implicit.type? ('card-' + _v.context.$implicit.type): (_co.type? ('card-' + _co.type): '')));
    const currVal_2:any = _co.isOpen(_v.context.$implicit.id);
    _ck(_v,3,0,currVal_0,currVal_1,currVal_2);
    const currVal_3:any = _v.context.$implicit.disabled;
    const currVal_4:any = (_v.context.$implicit.disabled? '-1': (null as any));
    const currVal_5:any = _co.isOpen(_v.context.$implicit.id);
    const currVal_6:any = (_co.isOpen(_v.context.$implicit.id)? _v.context.$implicit.id: (null as any));
    const currVal_7:any = _v.context.$implicit.disabled;
    _ck(_v,5,0,currVal_3,currVal_4,currVal_5,currVal_6,currVal_7);
    const currVal_8:any = _v.context.$implicit.title;
    _ck(_v,6,0,currVal_8);
  });
}
export function View_NgbAccordion_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵted((null as any),['\n    '])),(_l()(),i0.ɵand(16777216,
      (null as any),(null as any),1,(null as any),View_NgbAccordion_1)),i0.ɵdid(802816,
      (null as any),0,i1.NgForOf,[i0.ViewContainerRef,i0.TemplateRef,i0.IterableDiffers],
      {ngForOf:[0,'ngForOf']},(null as any)),(_l()(),i0.ɵted((null as any),['\n  ']))],
      (_ck,_v) => {
        var _co:i2.NgbAccordion = _v.component;
        const currVal_0:any = _co.panels;
        _ck(_v,2,0,currVal_0);
      },(null as any));
}
export function View_NgbAccordion_Host_0(_l:any):i0.ɵViewDefinition {
  return i0.ɵvid(0,[(_l()(),i0.ɵeld(0,(null as any),(null as any),2,'ngb-accordion',
      [['role','tablist']],[[1,'aria-multiselectable',0]],(null as any),(null as any),
      View_NgbAccordion_0,RenderType_NgbAccordion)),i0.ɵdid(2146304,(null as any),
      1,i2.NgbAccordion,[i3.NgbAccordionConfig],(null as any),(null as any)),i0.ɵqud(603979776,
      1,{panels:1})],(null as any),(_ck,_v) => {
    const currVal_0:boolean = !i0.ɵnov(_v,1).closeOtherPanels;
    _ck(_v,0,0,currVal_0);
  });
}
export const NgbAccordionNgFactory:i0.ComponentFactory<i2.NgbAccordion> = i0.ɵccf('ngb-accordion',
    i2.NgbAccordion,View_NgbAccordion_Host_0,{activeIds:'activeIds',closeOtherPanels:'closeOthers',
        destroyOnHide:'destroyOnHide',type:'type'},{panelChange:'panelChange'},([] as any[]));
